### PR TITLE
tests(legacy-javascript): cache variants, check in lockfile, …

### LINKED ISSUE #10566: test-legacy-javascript improvements
Right now @connorjclark is the `test-legacy-javascript` whisperer, but it would be good to spread that ability :) Some possible improvements:

- [ ] **lock test dependencies to a specific version**. It appears deps updates are the cause of changes like in https://github.com/GoogleChrome/lighthouse/pull/10555#issuecomment-611586368 and https://github.com/GoogleChrome/lighthouse/pull/10469#discussion_r393901338, which block landing a PR (and Travis may have different cached dependencies than the local machine, so re-running and pushing may not fix CI)
- [ ] **speed up the initial run(?)** It's not that common to touch something in `**/legacy-javascript/**`, but since Travis starts fresh every time it's a +6minute hit to CI when you do
- [ ] **improve interpretability of results** by the average contributor. Maybe readme explanation, simpler logged output, and specific assertions on top of the `git diff`?
  - e.g. for `sample_v2_round_trip.json` you know the diff should match `sample_v2.json`, and there are [unit tests](https://github.com/GoogleChrome/lighthouse/blob/a6fd756c991c425dc3f59deb9c75eda0af7c9bdb/lighthouse-core/test/report/proto-test.js) checking specific aspects of the generated json to call out the important parts. While if something goes `-3567 total` to `+4121 total` in `test-legacy-javascript`, it's unclear if committing the new count is going to cause an issue.
  - maybe this means separating the diagnostic tool(s) from the CI check? The diagnostic would still log lots of information, while the CI check would be more like, are some number of bytes found (or something more specific, not sure :)

### PR DESCRIPTION
**Pull request by connorjclark**
Fixes #10566

### CODE REVIEW

**Comment by connorjclark** [comment]
@brendankenny bump :) kinda-sorta a blocker on #10564

File: lighthouse-core/scripts/legacy-javascript/run.js [hunk 1]
```diff
@@ -8,11 +8,25 @@
...
 const {execFileSync} = require('child_process');
+const crypto = require('crypto');
 const LegacyJavascript = require('../../audits/legacy-javascript.js');
 const networkRecordsToDevtoolsLog = require('../../test/network-records-to-devtools-log.js');
-const VARIANT_DIR = `${__dirname}/variants`;
+
+// Create variants in a directory named-cached by contents of this script and the lockfile.
+// This folder is in the CI cache, so that the time consuming part of this test only runs if
+// the output would change.
+removeCoreJs(); // (in case the script was canceled halfway - there shouldn't be a core-js dep checked in.)
```
Comment (brendankenny) [inline lighthouse-core/scripts/legacy-javascript/run.js:21]: it does seem a little odd to control this here (both running here and not in `main()` and `yarn` commands inside this file in general), but I haven't reviewed the full control flow so I'll assume it makes sense :)

File: lighthouse-core/scripts/legacy-javascript/README.md [hunk 2]
```diff
@@ -12,6 +12,19 @@ node run.js
...
 `summary-sizes.json` - lists the size of each minified variant. Useful for understanding how many bytes each polyfill / transform adds.
 
+## Interpreting Results
+
+There are two outputs to this test:
+
+* summary-sizes.txt
+* summary-signals.json
+
+`summary-sizes.txt` lists each of the variants (grouped by type) and sorted by their byte size. This is mostly a diagnostic tool and changes in this can be ignored. This is checked in because whenever the lockfile is updated, `summary-sizes.txt` tells us things like if transforms are getting worse. And it is a quick reference for the relative weight of each of these transform/polyfills.
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply lighthouse-core/scripts/legacy-javascript/README.md:22]: yeah, I tried capturing that better with a rewording

File: lighthouse-core/scripts/legacy-javascript/run.js [hunk 3]
```diff
@@ -258,6 +272,36 @@ function makeSummary() {
   };
 }
 
+function createSummarySizes() {
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply lighthouse-core/scripts/legacy-javascript/run.js:275]: ehhhhh I wanted symmetry with `summary-signals`, i'll just leave it as is

**Pull request merged**